### ISSUE TITLE
[Bug Found & Fixed] Repo Custom `LayerNorm1d` Bug

### ISSUE THREAD

**Issue by OuYaozhong** [Pointcept/Pointcept#595]
Dear Pointcept Team,

I found a bug in the repo.

In [`pointcept/models/point_transformer/utils.py`](https://github.com/Pointcept/Pointcept/blob/d74c646db6abec569d0f23e0c34e7ddfce142789/pointcept/models/point_transformer/utils.py#L7), you defined a `LayerNorm1d` class that inherits from `nn.BatchNorm1d`, named as `LayerNorm1d`, to achieve normalization of the `dim = -1` instead of `dim = -2` by default of the `nn.BatchNorm1d`.

But I found this will raise a problem when using `sync_bn = True`.

Because when enable `sync_bn`, the [`nn.SyncBatchNorm.convert_sync_batchnorm`](https://github.com/Pointcept/Pointcept/blob/d74c646db6abec569d0f23e0c34e7ddfce142789/pointcept/engines/train.py#L257-L258) located in `pointcept/engines/train.py` will transfer all the classes inherited from `_BatchNorm` into `torch.nn.SyncBatchNorm`, which will make the size of `num_features` mismatched since the action is no longer customized one we want.

I just modified a feasible one that can achieve the original goal but will compatible for the `sync_bn = True` cases, as following one.

```
import torch
import torch.nn as nn

torch.nn.LayerNorm


# class LayerNorm1d(nn.BatchNorm1d):
#     def forward(self, input: torch.Tensor) -> torch.Tensor:
#         return (
#             super()
#             .forward(input.transpose(1, 2).contiguous())
#             .transpose(1, 2)
#             .contiguous()
#         )


class LayerNorm1d(nn.Module):

    def __init__(
        self,
        num_features: int,
        eps: float = 1e-5,
        momentum: float | None = 0.1,
        affine: bool = True,
        track_running_stats: bool = True,
        device=None,
        dtype=None,
    ) -> None:
        super().__init__()
        self.bn = nn.BatchNorm1d(
            num_features=num_features,
            eps=eps,
            momentum=momentum,
            affine=affine,
            track_running_stats=track_running_stats,
            device=device,
            dtype=dtype
        )

    def forward(self, input: torch.Tensor) -> torch.Tensor:
        return (
            self.bn(input.transpose(1, 2).contiguous())
            .transpose(1, 2)
            .contiguous()
        )
```

Above code has been successfully run under `sync_bn = True` in my machine.

For the one who met the same problem and the repo owners, as one of references, to correct it.

**Comment by Gofinge** [comment]
Thank you, I think you are correct. Would you like to open a PR to contribution to the fix of this issue?
